ksql-engine/src/test/java/io/confluent/ksql/function/udf/url/UrlDecodeParamKudfTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.confluent.ksql.function.udf.url;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class UrlDecodeParamKudfTest {
```

Comment:
this one throws an exception on bad encoding right? We should test that path...

Reply:
not exactly - the "bad encoding" only throws if you pass in an invalid charset (e.g. UTF7). I don't think that java will actually let you compile an unknown encoding :)